hedera-node/test-clients/src/main/java/com/hedera/services/bdd/spec/HapiSpec.java

```diff
@@ -189,6 +192,9 @@ public static void setStakerIds(@Nullable final Set<AccountID> stakerIds) {
 
     public static final ThreadLocal<HederaNetwork> TARGET_NETWORK = new ThreadLocal<>();
     public static final ThreadLocal<BlockNodeNetwork> TARGET_BLOCK_NODE_NETWORK = new ThreadLocal<>();
+    public static final ThreadLocal<HederaNetwork> PRIMARY_NETWORK = new ThreadLocal<>();
+    public static final ThreadLocal<HederaNetwork> SECONDARY_NETWORK = new ThreadLocal<>();
+    public static final ThreadLocal<DualNetworkEnvironment> DUAL_ENVIRONMENT = new ThreadLocal<>();
```

Comment:
`TARGET_NETWORK`, as a global singleton, already makes it challenging to handle encapsulating network properties and behaviors. Adding more thread locals doesn't seem like the best option